timeseries/src/autogluon/timeseries/models/abstract/abstract_timeseries_model.py

```diff
@@ -255,16 +257,22 @@ def fit(
         """
         start_time = time.monotonic()
         self.initialize(**kwargs)
+
         if self.target_scaler is not None:
             train_data = self.target_scaler.fit_transform(train_data)
-
         if self.covariate_regressor is not None:
-            train_data = self.covariate_regressor.fit_transform(
+            regressor_time_limit = self._regressor_fit_time_fraction * time_limit if time_limit is not None else None
+            self.covariate_regressor.fit(
                 train_data,
-                time_limit=0.5 * time_limit if time_limit is not None else None,
+                time_limit=regressor_time_limit,
+                verbosity=kwargs.get("verbosity", 2) - 1,
             )
 
-        train_data = self.preprocess(train_data, is_train=True)
+        if self._get_tags()["can_use_train_data"]:
+            if self.covariate_regressor is not None:
+                train_data = self.covariate_regressor.transform(train_data)
+            train_data = self.preprocess(train_data, is_train=True)
```

Comment:
is it now the case that `train_data` can be undefined during the `super().fit()` call? i.e., should we default it to `train_data=None` above?

Reply:
The `train_data` will always be a `TimeSeriesDataFrame`. However, in case `{"can_use_train_data": False}`, we will pass an un-preprocessed dataframe to `super()._fit()` since the dataframe anyway won't be used inside (to save time on preprocessing).